spict/R/utility.R

```diff
@@ -536,3 +536,216 @@ get.cov <- function(rep, parname1, parname2, cor=FALSE){
+            }
+        }
+        inpout$nobsI <- length(inpout$obsI[[1]])
+    }
+
+    ## effort
+    if(length(inpin$obsE) > 0){
+        inds <- get.inds(inpin$timeE,mintime,maxtime)
+        inpout$obsE <- inpin$obsE[inds]
+        inpout$timeE <- inpin$timeE[inds]
+        inpout$stdevfacE <- inpin$stdevfacE[inds]
+        inpout$dte <- inpin$dte[inds]
+    }
+
+    timeobsall <- sort(c(inpout$timeC, inpout$timeC + inpout$dtc,
+                         unlist(inpout$timeI),
+                         inpout$timeE, inpout$timeE + inpout$dte))
+    if (length(timeobsall) == 0) stop("All yars are outside the range ", mintime, "-", maxtime)
+
+    # Time point to predict catches until
+    inpout$timepredc <- max(timeobsall)
+    inpout$timepredi <- max(timeobsall)
+
+    # This may give a problem if effort data has later time points than catches or index
+    if (inpout$nobsE > 0 & sum(inpout$nobsI) > 0){
+        if (max(inpout$timeE) > max(unlist(inpout$timeI), inpout$timeC)){
+            stop('Effort data must overlap temporally with index or catches')
+        }
+    }
+    if ("eulertype" %in% names(inpout)){
+        if (inpout$eulertype == 'hard'){
+            # Hard Euler discretisation
+            if (inpout$start.in.first.data.point){
+                time <- seq(min(timeobsall),
+                            max(inpout$timepredi, inpout$timepredc+inpout$dtpredc),
+                            by=inpout$dteuler)
+            } else {
+                time <- seq(floor(min(timeobsall)),
+                            max(inpout$timepredi, inpout$timepredc+inpout$dtpredc),
+                            by=inpout$dteuler)
+            }
+            inpout$time <- time
+        }
+        if (inpout$eulertype == 'soft'){
+            # Include times of observations (including dtc)
+            time <- seq(ceiling(min(timeobsall)),
+                        max(inpout$timepredi, inpout$timepredc+inpout$dtpredc),
+                        by=inpout$dteuler)
+            inpout$time <- sort(unique(c(timeobsall, time)))
+        }
+        if (!inpout$eulertype %in% c('soft', 'hard'))
+            stop('inp$eulertype must be either "soft" or "hard"!')
+    }
+
+    inpout$dt <- inpout$dt[inpin$time %in% inpout$time]
+    inpout$ns <- length(inpout$time)
+
+    inpout$ffacvec <- inpout$ffacvec[inpin$time %in% inpout$time]
+    inpout$fconvec <- inpout$fconvec[inpin$time %in% inpout$time]
+    inpout$ini$logF <- inpout$ini$logF[inpin$time %in% inpout$time]
+    inpout$ini$logu <- inpout$ini$logu[,(inpin$time %in% inpout$time)]
+    inpout$ini$logB <- inpout$ini$logB[inpin$time %in% inpout$time]
+    inpout$ini$logmre <- inpout$ini$logmre[inpin$time %in% inpout$time]
+    inpout$ini$SARvec <- inpout$ini$SARvec[inpin$time %in% inpout$time]
+    inpout$MSYregime <- inpout$MSYregime[inpin$time %in% inpout$time]
+    inpout$regimeIdx <- NULL
+    inpout$manstart <- NULL
+    inpout$ir <- NULL
+    inpout$ini$logr <- NULL
```

Comment:
If we are cutting the time using only mintime, why do we remove the `manstart`.

I am not sure what the others are, regime shift options?

Reply:
If we are cutting using only `mintime`, we could keep `manstart`. Only if `maxtime` is used it won't make sense any more. 
`inp$ir` indicates the different regimes as numeric values. `inp$ini$logr` is the intrinsic growth rate from the traditional surplus production model. If `logm` is used, `logr` is not being used.

Reply:
add documentation that management and regimes are set to `NULL`. If used user needs to reset them after `shorten.inp`.

Reply:
keep `manstart` if only `mintime` used

Reply:
run `check.inp` again or copy defaults for these variables from `check.inp`